controller/api/models.py

```diff
@@ -292,6 +292,7 @@ def _start_containers(self, to_add):
         if any(c.state != 'created' for c in to_add):
             err = 'aborting, failed to create some containers'
             log_event(self, err, logging.ERROR)
+            self._destroy_containers(to_add)
```

Comment:
what ever the containers create objects that are there in to_add will get destroyed. Yes, I agree that this should be handled better in scale operation rather than start_containers. But this is a temporary fix just to make sure we are rolling back the containers if resources are not satisfied by scale requirements. Any ways there is plan to achieve this in https://github.com/deis/deis/issues/3850, but this wont be ready for at-least 1 month.